File: mobile/src/main/java/net/stecky/puzzlemethis/EventAdapter.java
package net.stecky.puzzlemethis;

import android.content.Context;
import android.support.v7.widget.RecyclerView;
import android.view.ViewGroup;

import net.stecky.puzzlemethis.events.Event;
import net.stecky.puzzlemethis.viewholders.BadBehaviorViewHolder;
import net.stecky.puzzlemethis.viewholders.BaseEventViewHolder;

import java.util.List;

/**
 * Created by sb185219 on 11/8/2017.
 */

public class EventAdapter extends RecyclerView.Adapter<BaseEventViewHolder>
{
    private Context mContext;
    private List<Event> eventList;


    public EventAdapter(Context mContext, List<Event> eventList)
    {
        this.mContext = mContext;
        this.eventList = eventList;
    }

    @Override
    public BaseEventViewHolder onCreateViewHolder(ViewGroup parent, int viewType)
    {
        switch(viewType)
        {
            case 0: return new BadBehaviorViewHolder(parent);
        }

        return null;
    }

    @Override
    public void onBindViewHolder(BaseEventViewHolder holder, int position)
    {
        Event event = eventList.get(position);

        switch(holder.getItemViewType())
        {
            case 0: holder.populateValuesFromEvent(event);
        }
    }

    @Override
    public int getItemCount()
    {
        return eventList.size();
    }
}

File: mobile/src/main/java/net/stecky/puzzlemethis/EventAdapter.kt
package net.stecky.puzzlemethis

import android.content.Context
import android.support.v7.widget.RecyclerView
import android.view.ViewGroup
import net.stecky.puzzlemethis.events.BaseEventViewHolder
import net.stecky.puzzlemethis.events.Event
import net.stecky.puzzlemethis.events.behaviors.BadBehaviorViewHolder
import net.stecky.puzzlemethis.events.health.MedicineAdministrationViewHolder

/**
 * Created by sb185219 on 11/14/2017.
 */
class EventAdapter(private val mContext: Context, private val eventList: List<Event>) : RecyclerView.Adapter<BaseEventViewHolder>()
{
    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): BaseEventViewHolder?
    {
        when(viewType)
        {
            1 -> return BadBehaviorViewHolder(parent)
            2 -> return MedicineAdministrationViewHolder(parent)
        }

        return null
    }

    override fun onBindViewHolder(holder: BaseEventViewHolder, position: Int)
    {
        holder.onBindViewHolder(event = eventList[position])
    }

    override fun getItemCount(): Int
            = eventList.size

    override fun getItemViewType(position: Int): Int
    {
        return eventList[position].type.ordinal
    }
}